cmd/minishift/cmd/profile/profile_set.go

```diff
@@ -41,26 +37,47 @@ func runProfile(cmd *cobra.Command, args []string) {
 	validateArgs(args)
 	profileName := args[0]
 
+	err := unsetCurrentOcContext(profileActions.GetActiveProfile())
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+
 	if !cmdUtil.IsValidProfile(profileName) {
```

Comment:
why do you use negated logic in the if statement, especially if you have a if/else? Negated logic is always harder to work out in our brain, since you have to put some mental effort into working out what it actually means. Given that you have both branches either way, why not 'if !cmdUtil.IsValidProfile(profileName)'? Less mental load.

Reply:
@hferentschik I agree. @anjannath Lets fix this in this PR.

Reply:
ack @LalatenduMohanty Updating the PR with the requested changes.